### PR TITLE
12 profile edit

### PR DESCRIPTION
**Pull request by rostom44**
Structure de edit profile front
backend terminer et lier avec le front

### CODE REVIEW

**Review by MaksimLsk — APPROVED** [review]
good

File: client/src/components/Profile/UserInfo.jsx [hunk 1]
```diff
@@ -34,9 +35,11 @@ export default function UserInfo({ user }) {
             <p>{city}</p>
           </div>
         </div>
-        <button type="button" className="button">
-          <h3>Edit</h3>
-        </button>
+        <Link to={`/profile/${user.id}/edit`}>
+          <button type="button" className="button">
+            <h3>Edit</h3>
```
Comment (benrcd) [inline client/src/components/Profile/UserInfo.jsx]: `h3 `tag aren't allowed within `button`

**Review by benrcd — CHANGES_REQUESTED** [review]

**Comment by benrcd** [comment]
We also need the wireframe (maquette) that goes with the page

File: server/database/models/userRepository.js [hunk 2]
```diff
@@ -72,6 +72,32 @@ class userRepository extends AbstractRepository {
 
     return user;
   }
+
+  async update(user) {
+    if (
+      !user.firstname ||
```
Comment (wildmatthieu) [inline server/database/models/userRepository.js]: condition à expliciter : 
!user.firstname === ??? || 
!user.lastname === ??? ||
etc...

**Review by wildmatthieu — CHANGES_REQUESTED** [review]

**Review by wildmatthieu — APPROVED** [review]

**Pull request merged**